### PR TITLE
feat: support ecotone

### PR DESCRIPTION
**Pull request by chokobole**
# Description

This PR updates fee collection logic for Optimism Ecotone.

See https://specs.optimism.io/protocol/exec-engine.html#ecotone-l1-cost-fee-changes-eip-4844-da

### CODE REVIEW

File: bus-mapping/src/state_db.rs [hunk 1]
```diff
@@ -331,16 +331,37 @@ impl StateDB {
     pub fn compute_l1_fee(
         &self,
         l1_base_fee: Word,
-        l1_fee_overhead: Word,
-        l1_fee_scalar: Word,
+        l1_base_fee_scalar: Word,
+        l1_blob_base_fee: Word,
         rollup_data_gas_cost: u64,
     ) -> Result<Word, Error> {
+        use eth_types::ToBigEndian;
```
Comment (dongchangYoo) [inline bus-mapping/src/state_db.rs:338]: did you intend to insert this line here?

File: bus-mapping/src/state_db.rs [hunk 2]
```diff
@@ -331,16 +331,37 @@ impl StateDB {
...
+        debug_assert!(!l1_blob_base_fee.is_zero());
+
+        let l1_base_fee_scalar_bytes = l1_base_fee_scalar.to_be_bytes();
+        const L1_BASE_FEE_SCALAR_OFFSET: usize = 16;
+        let l1_base_fee_scalar = u32::from_be_bytes([
+            l1_base_fee_scalar_bytes[L1_BASE_FEE_SCALAR_OFFSET],
+            l1_base_fee_scalar_bytes[L1_BASE_FEE_SCALAR_OFFSET + 1],
+            l1_base_fee_scalar_bytes[L1_BASE_FEE_SCALAR_OFFSET + 2],
+            l1_base_fee_scalar_bytes[L1_BASE_FEE_SCALAR_OFFSET + 3],
+        ]);
```
Comment (dongchangYoo) [inline bus-mapping/src/state_db.rs]:
```suggestion
        let l1_base_fee_scalar = u32::from_be_bytes(
            l1_base_fee_scalar_bytes[L1_BASE_FEE_SCALAR_OFFSET..(L1_BASE_FEE_SCALAR_OFFSET + 4)]
                .try_into()
                .expect("slice with incorrect length"),
        );
```

File: bus-mapping/src/state_db.rs [hunk 3]
```diff
@@ -331,16 +331,37 @@ impl StateDB {
...
+        const L1_BLOB_BASE_FEE_SCALAR_OFFSET: usize = 12;
+        let l1_blob_base_fee_scalar = u32::from_be_bytes([
+            l1_base_fee_scalar_bytes[L1_BLOB_BASE_FEE_SCALAR_OFFSET],
+            l1_base_fee_scalar_bytes[L1_BLOB_BASE_FEE_SCALAR_OFFSET + 1],
+            l1_base_fee_scalar_bytes[L1_BLOB_BASE_FEE_SCALAR_OFFSET + 2],
+            l1_base_fee_scalar_bytes[L1_BLOB_BASE_FEE_SCALAR_OFFSET + 3],
+        ]);
```
Comment (dongchangYoo) [inline bus-mapping/src/state_db.rs]:
```suggestion
        let l1_blob_base_fee_scalar = u32::from_be_bytes(
            l1_base_fee_scalar_bytes
                [L1_BLOB_BASE_FEE_SCALAR_OFFSET..(L1_BLOB_BASE_FEE_SCALAR_OFFSET + 4)]
                .try_into()
                .expect("slice with incorrect length"),
        );
```

**Review by dongchangYoo — CHANGES_REQUESTED** [review]

**Comment by dongchangYoo** [comment]
please change minor version of the zkevm-circcuit, see `zkevm-circuits/src/version.rs`

File: mock/src/test_ctx.rs [hunk 4]
```diff
@@ -428,32 +428,35 @@ pub mod helpers {
...
         // batcherHash
         calldata.extend(padding!(vec![
             0x3c, 0x44, 0xcd, 0xdd, 0xb6, 0xa9, 0x00, 0xfa, 0x2b, 0x58, 0x5d, 0xd2, 0x99, 0xe0,
             0x3d, 0x12, 0xfa, 0x42, 0x93, 0xbc
         ]));
-        // l1 fee overhead: 2100
-        calldata.extend(padding!(vec![0x08, 0x34]));
-        // l1 fee scalar: 1000000
-        calldata.extend(padding!(vec![0x0f, 0x42, 0x40]));
         // validator reward scalar: 2000
```
Comment (fakedev9999) [inline mock/src/test_ctx.rs]:
```suggestion
        // validator reward scalar: 5000
```

**Review by fakedev9999 — CHANGES_REQUESTED** [review]

File: bus-mapping/src/state_db.rs (hunk 1 above)
Comment (chokobole) [reply bus-mapping/src/state_db.rs:338]: This is needed , without it, we have this error.

```rust
error[E0599]: no method named `to_be_bytes` found for struct `eth_types::U256` in the current scope
   --> bus-mapping/src/state_db.rs:342:59
    |
342 |         let l1_base_fee_scalar_bytes = l1_base_fee_scalar.to_be_bytes();
    |                                                           ^^^^^^^^^^^ method not found in `eth_types::U256`
```

**Review by dongchangYoo — APPROVED** [review]
LGTM

**Review by fakedev9999 — APPROVED** [review]
LGTM

**Pull request merged**